src/rust/src/error.rs

```diff
@@ -0,0 +1,45 @@
+use flatbuffers::InvalidFlatbuffer;
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug)]
+pub enum Error {
+    Malformed(&'static str),
```

Comment:
Are there many different error texts? @nyurik argued that we shouldn't use freetext messages for errors at all.

Reply:
@pka I think `static str` is better than `String` (no allocation, much faster errors, esp when no error logging is needed), but I also think that a dedicated error type is better than `static str` because the user can make decisions based on the error type, and the compiler will ensure the stability of those errors (as oppose to doing brittle error string comparison).

Reply:
I think freetext messages can be OK if you don't intend for the user to handle the specific texts differently.

That said, I think in this case Malformed was doing more work than it should. See b7e7f73

Are the unwrap's controversial? We're already asserting the same things that guarantee the invariant within the `generate_bounds` method so I don't think it introduces any crash in practice. I could leverage the typesystem to return some kind of "NonEmptyVec", but I'm not sure it's worth it.